lib/workbox/options.js

```diff
@@ -21,6 +21,11 @@ function getOptions (nuxt, pwa) {
     options.publicPath = publicPath
   }
 
+  // revision
+  if (!options.cacheOptions.revision) {
+    options.cacheOptions.revision = randomString(12)
+  }
```

Comment:
If no revision is set, browser complains about precaching file without revision:
![image](https://user-images.githubusercontent.com/2473784/96711986-b1949f80-139e-11eb-82b6-08f42a6d7435.png)

I think that adding some randomized revision as a default would be beneficial.
I tried to get random version from `nuxt` environmental variables, but couldn't find it.

Reply:
I think we have to pass `null` instead for revision info of `_nuxt/` assets because we always add hash to URL for long-term caching:

>   {url: '/scripts/app.0d5770.js', revision: null},
> For the second and third object in the example above, the revision property is set to null. This is because the revisioning information is in the URL itself, which is generally a best practice for static assets.

Reply:
Sure, for script assets that's true. But this case is about `.html` files, which do not have any hash in name.
Once they are precached, you cannot do anything to update them, other than unregistering and registering service worker.

Reply:
The same applies to offline page (`404.html`) and start url for PWA (`/?standalone=true` by default). 
Once they get precached, PWA stops being updated and offline page too.

This is a behavior I observed in my own application - when installed app as a standalone PWA application it refused to be updated even if completely new deploy was made.

Reply:
I see you are right. But the problem is that generating revision on build-time only works for static generated websites. If response of a SSR page changes, workbox still refuses to update..

Reply:
I see you are right. But the problem is that generating revision on build-time only works for static generated websites. If response of a SSR page changes, workbox still refuses to update..

Reply:
If precache would work with `NetworkFirst`, everything would be fine.
But from what I've read, it always work as `CacheFirst`.

Maybe it's a good idea to disable precaching at all? Or replace it with manually fetching those pages in `sw.js` file and caching them with standard `NetworkFirst` cache, instead of depending on `workbox` standard precache?

Reply:
@kedrzu This PR looks good after and partially fixing issue. But indeed, precaching is only `CacheFirst` (https://github.com/GoogleChrome/workbox/issues/1767). 

I will try to implement new `offlineRoutes` option for (NF or SWR) strategy :)

Reply:
#406